CATTY-530 Modify Swift extension for ProjectDetailStoreViewController

dwildr opened the pull request:
https://jira.catrob.at/browse/CATTY-530

The following methods have been moved from ProjectDetailStoreViewController.m to ProjectDetailStoreViewControllerExtension.swift:
loadProject
createViewForProject

And the following method has been removed: reloadWithProject

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

Bomml reviewed (approved):
Nice work!

m-herold reviewed (changes requested):
When I try to open a project in the Store (e.g., any in "Most downloaded"), then Catty crashes. Do your changes work locally?

dwildr commented:
It definitely works (and worked) in the simulator locally. What's interesting is, that I now rebased the PR with the current develop branch and then the `ProjectDetailStoreViewControllerExtensionTests` failed (which worked when I initially opened the PR). I fixed the problem and maybe this was the same problem you where getting?

I would have liked to test it on my physical device, but since I have a new iPhone now, I am no longer able to install the build. (I sent you a DM with my new UDID)

m-herold reviewed (approved):
Thanks @dwildr, everythings works perfectly now!
